### ISSUE TITLE
KeyError when loading particle-level data from specified MCST halo

### ISSUE THREAD

**Issue by jrspreng** [cbyrohl/scida#185]
When trying to load data from a given MCST halo a key error is raised. It seems like scida is trying to load fields that exist in TNG but not in MCST (here: ElectronAbundance).

Code used:

```
sim = sc.load('/virgotng/mpia/MCST/L35n2160TNG_h10677_z3_L14_ST8', units=False)
snapshot = sim.get_dataset(redshift=4)
groupid = snapshot.data["PartType1"]["GroupID"]
particles = snapshot.return_data(haloID=0)
```

Error:

```
`---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Cell In[3], line 4
      2 snapshot = sim.get_dataset(174)
      3 groupid = snapshot.data["PartType1"]["GroupID"]
----> 4 particles = snapshot.return_data(haloID=0)

File /vera/u/jonsp/scida/src/scida/interface.py:443, in Selector.__call__.<locals>.newfn(*args, **kwargs)
    440 self.data = args[0].data.copy_skeleton()
    441 # deepdictkeycopy(self.data_backup, self.data)
--> 443 self.prepare(*args, **kwargs)
    444 if self.keys is None:
    445     raise NotImplementedError(
    446         "Subclass implementation needed for self.keys!"
    447     )

File /vera/u/jonsp/scida/src/scida/customs/arepo/selector.py:64, in ArepoSelector.prepare(self, *args, **kwargs)
     62 objtype = "subhalo" if subhalo_id is not None else "halo"
     63 if idx is not None:
---> 64     self.select_group(snap, idx, objtype=objtype)
     65 elif unbound is True:
     66     self.select_unbound(snap)

File /vera/u/jonsp/scida/src/scida/customs/arepo/selector.py:136, in ArepoSelector.select_group(self, snap, idx, objtype)
    134         if hasattr(length, "magnitude"):
    135             length = length.magnitude
--> 136         for k, v in self.data_backup[p].items():
    137             self.data[p][k] = v[offset : offset + length]
    138 snap.data = self.data

File /vera/u/jonsp/scida/src/scida/fields.py:359, in <genexpr>(.0)
    340 def items(self, withrecipes=True, withfields=True, evaluate=True):
    341     """
    342     Return a list of tuples for keys/values in the container.
    343 
   (...)
    356 
    357     """
    358     return (
--> 359         (k, self._getitem(k, evaluate_recipe=evaluate))
    360         for k in self.keys(withrecipes=withrecipes, withfields=withfields)
    361     )

File /vera/u/jonsp/scida/src/scida/fields.py:656, in FieldContainer._getitem(self, key, force_derived, update_dict, evaluate_recipe)
    654 if not evaluate_recipe:
    655     return self._fieldrecipes[key]
--> 656 field = self._instantiate_field(key)
    657 if update_dict:
    658     self._fields[key] = field

File /vera/u/jonsp/scida/src/scida/fields.py:702, in FieldContainer._instantiate_field(self, key)
    694     kwargs.update(
    695         **{
    696             k: v
   (...)
    699         }
    700     )
    701 # finally, instantiate field
--> 702 field = func(self, **kwargs)
    703 if self.withunits and units is not None:
    704     if not hasattr(field, "units"):

File /vera/u/jonsp/scida/src/scida/customs/arepo/dataset.py:1662, in Temperature(arrs, ureg, **kwargs)
   1659     m_p = m_p * ureg.g
   1660     k_B = k_B * ureg.erg / ureg.K
-> 1662 xe = arrs["ElectronAbundance"]
   1663 u_internal = arrs["InternalEnergy"]
   1665 mu = 4 / (1 + 3 * xh + 4 * xh * xe) * m_p

File /vera/u/jonsp/scida/src/scida/fields.py:445, in FieldContainer.__getitem__(self, key)
    444 def __getitem__(self, key):
--> 445     return self._getitem(key)

File /vera/u/jonsp/scida/src/scida/fields.py:661, in FieldContainer._getitem(self, key, force_derived, update_dict, evaluate_recipe)
    659     return field
    660 else:
--> 661     raise KeyError("Unknown field '%s'" % key)

KeyError: "Unknown field 'ElectronAbundance'"`
```

**Environment:**
python version: 3.9.20
scida version: 0.3.3
pint version: 0.24.4
dask version: 2024.8.0

**Comment by cbyrohl** [comment]
Thanks!, fixed with #186. Let me know if that works for you.

**Comment by jrspreng** [comment]
Works as it should now. Thanks!
